File: Python/OOP/Inheritance/Single inehritance (example 3).py
class Teacher:
    teacher_name = "Guido van Rossum"
    subject = "Python Programming"

    def teacherInfo(self):
        print("Teacher info: \n--------------")
        print("Name   :", self.teacher_name)
        print("Subject:", self.subject)


class Student(Teacher):
    student_name = "Abdullah Al Naiem"
    id = 123456

    def studentInfo(self):
        print("\nStudent info: \n--------------")
        print("Name:", self.student_name)
        print("ID  :", self.id)
        print("Class of:", self.subject)


class singleInheritance:
    get = Student()
    get.teacherInfo()
    get.studentInfo()

File: Java/OOP/Inheritance/Single inehritance (example 3).java
class Teacher {
    String teacher_name = "James Gosling";
    String subject = "Java Programming";

    void teacherInfo() {
        System.out.println("Teacher info: \n--------------");
        System.out.println("Name   : " + teacher_name);
        System.out.println("Subject: " + subject);
    }
}

class Student extends Teacher {
    String student_name = "Abdullah Al Naiem";
    int id = 123456;

    void studentInfo() {
        System.out.println("\nStudent info: \n--------------");
        System.out.println("Name: " + student_name);
        System.out.println("ID  : " + id);
        System.out.println("Class of: " + subject);
    }
}

class singleInhertance {
    public static void main(String[] args) {
        Student get = new Student();
        get.teacherInfo();
        get.studentInfo();
    }
}